deku-derive/src/macros/deku_read.rs

```diff
@@ -38,6 +38,23 @@ fn emit_struct(input: &DekuData) -> Result<TokenStream, syn::Error> {
 
     let initialize_struct = super::gen_struct_init(is_named_struct, internal_fields);
 
+    let check_magic = if let Some(magic) = &input.magic {
```

Comment:
Similarly, check_magic should also be available for [emit_enum](https://github.com/sharksforarms/deku/blob/22e271d421574cf9ebc6f9e00c42feb0978ecb70/deku-derive/src/macros/deku_read.rs#L249)